samples/download-preplanned-map-area/src/main/java/com/esri/arcgismaps/sample/downloadpreplannedmaparea/screens/DownloadPreplannedMapAreaScreen.kt

```diff
@@ -0,0 +1,197 @@
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentSize(), onDismissRequest = {
+                            showBottomSheet = false
+                        }, sheetState = sheetState
+                    ) {
+                        DownloadPreplannedMap(
+                            mapViewModel::showOnlineMap,
+                            mapViewModel::downloadOrShowOfflineMap,
+                            mapViewModel.preplannedMapAreaInfoFlow
+                        )
+                    }
+                }
+            }
+            mapViewModel.messageDialogVM.apply {
+                if (dialogStatus) {
+                    MessageDialog(
+                        title = messageTitle, description = messageDescription, onDismissRequest = ::dismissDialog
+                    )
+                }
+            }
+        },
+        floatingActionButton = {
+            if (!showBottomSheet) {
+                FloatingActionButton(modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { showBottomSheet = true }) {
+                    Icon(
+                        Icons.Filled.Settings, contentDescription = "Select map"
+                    )
+                }
+            }
+        })
+}
+
+@Composable
+fun DownloadPreplannedMap(
+    showOnlineMap: () -> Unit,
+    downloadOrShowMap: (PreplannedMapAreaInfo) -> Unit,
+    preplannedMapAreaInfoFlow: StateFlow<List<PreplannedMapAreaInfo>>
+) {
+    Column(
+        modifier = Modifier
+            .wrapContentSize()
+            .background(MaterialTheme.colorScheme.background)
+            .padding(12.dp)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.Start,
+    ) {
+        Text(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(16.dp),
+            text = "Select map",
+            style = MaterialTheme.typography.titleLarge
+        )
+        Card(modifier = Modifier.wrapContentSize()) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(8.dp)
+                    .clickable { showOnlineMap() },
+                horizontalArrangement = Arrangement.spacedBy(4.dp),
+                verticalAlignment = CenterVertically,
+            ) {
+                Text("Web map (online)")
+            }
+        }
+        Text(text = "Preplanned map areas", modifier = Modifier.padding(top = 16.dp, start = 8.dp, bottom = 4.dp))
+        Card(modifier = Modifier.wrapContentSize()) {
+            preplannedMapAreaInfoFlow.collectAsState().value.forEach {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(8.dp)
+                        .clickable {
+                            downloadOrShowMap(it)
+                        },
```

Comment:
I would suggest move to add the `clickable` before applying the `padding` so that the entire row is clickable (plus the padding)
```suggestion
                        .clickable { downloadOrShowMap(it) }
                        .padding(8.dp),
```